### PR TITLE
Documentation Rewrite

### PR DESCRIPTION
**Pull request by smcdougall**
Thought of making this a new Issue, but since I've made some commits already (notes and minor fixes), I decided to just PR, as it'll need to be one eventually.

Currently, this contains some unmerged commits from #352.

New commits start at `Added Improvement Notes to Docs. Minor fixes and improvements.` https://github.com/picocms/Pico/commit/3a76be4356ad418855ee5c320728b75e58448ad2

The idea here is going to be a full (or close to) rewrite of the existing Documentation into smaller, more in-depth pages.

Since the plan is to redesign the docs around this new style, it'll only really be merge-able when _all_ the work is done.  For this reason I've decided to make it one PR and not do a PR per-page.  If this changes, or I write a decent, standalone page, we can always merge them in on a case-by-case basis.

Discussion on this topic was previously started on #352.

Here's some quotes from that thread:
- Page Topics

> - An "Installing Pico" page that covers, in-depth, the concepts of installing and configuring Pico, your webserver (well, links to external resources anyway) and etc. Everything up until you're looking at Pico's sample content.
> - A "Creating Content" page that goes over YAML and Markdown.
>   Perhaps a general, "Usage" type page that covers things like how URLs work, how to lay out and organize your website, separating content and assets, more about config options, etc.
> - Customization: An in-depth guide to installing community themes, creating your own, understanding Twig, Pico variables, everything. Not plugins though.
> - A separate page for Plugins (and plugin development).
> - Maybe a combination of "Getting Help" and "Contributing" (or "Getting Involved"). It should be a generally community-focused page. On that note, I think it would be great if we eventually had a place for non-development, non-bug related community discussions. I don't want to say a "forum" because that feels like a really outdated notion, but a "place" to develop a community nonetheless.
- Table of Contents

> When on the Overview Page:
> 
> ```
> * Installing Pico
> * Creating Content
> * Customization
> * Getting Help
> ```
> 
> When on Creating Content Page:
> 
> ```
> * Installing Pico
> * Creating Content
>    * YAML
>    * Markdown
>    * Etc
> * Customization
> * Getting Help
> ```

---
- Advanced asset handling guide https://github.com/picocms/Pico/pull/368#issuecomment-243415712
- Definition Lists for Twig Terms https://github.com/picocms/Pico/pull/358#discussion_r83863940
## Cookbook Examples
- ~~https://github.com/picocms/Pico/pull/334#issuecomment-255223633~~

### CODE REVIEW

**Comment by PhrozenByte** [comment]
Note for myself: Use https://github.com/picocms/Pico/pull/359/files/a48924bbc2d3ce208c86554d6c05dae667132d8e..HEAD to get a diff of all changes except for the changes of #352

@smcdougall: You can btw cherry pick the additional commits on a clean branch using `git cherry-pick 3a76be4^..e28a3da` if you want to :wink:

File: _docs/customization.md [hunk 1]
```diff
@@ -60,18 +66,32 @@ Pages can be used like the following:
...
 
+* These filters feel *way* to complicated to just casually mention at the end of the variables list.
+
 You can use different templates for different content files by specifying the `Template` meta header. Simply add e.g. `Template: blog-post` to a content file and Pico will use the `blog-post.twig` file in your theme folder to render the page.
 
+* I wish I had a use for this function, really, but why would I want to rewrite my entire template when I can just have conditional sections using `if`.  I think the only way I'd use it would be if someone *else* provided a theme that had multiple templates built in.  If I were building it myself I'd never elect to rewrite everything, I'd probably just use my own meta variable.
```
Comment (PhrozenByte) [inline _docs/customization.md]: You really should use different templates for different content categories, e.g. using the same template for blog posts, the blog overview and the landing page makes the template nearly impossible to understand and maintain. Use includes for common parts of a website (e.g. header and footer). See the `_layouts` and `_includes` folders of Pico's website :wink:

File: _docs/customization.md [hunk 2]
```diff
@@ -60,18 +66,32 @@ Pages can be used like the following:
...
+* And the website.  These references are all inconsistant.  Some say Wiki, some say website (I think).  Both places should probably be mentioned every time.
+  * Or better yet, lets kill all references to the wiki and just make the wiki more apparent on the Customization page.  Give it it's own box in with the themes and plugins or at the very least a line underneath them (but not a footnote like it has now.)
+
 ### Plugins
 
 #### Plugins for users
 
 Officially tested plugins can be found at [{{ site.github.url }}/customization/][OfficialPlugins], but there are many awesome third-party plugins out there! A good start point for discovery is [our Wiki][WikiPlugins].
 
+* *Are* these officially tested?  And is this list current?
```
Comment (PhrozenByte) [inline _docs/customization.md]: No. But I'm planning to do so :smile:

File: _docs/run.md [hunk 3]
```diff
@@ -9,7 +9,16 @@ nav: 3
 You have nothing to consider specially, simply navigate to your Pico install using your favorite web browser. Pico's default contents will explain how to use your brand new, stupidly simple, blazing fast, flat file CMS.
 
 ### You don't have a web server?
-Starting with PHP 5.4 the easiest way to try Pico is using [the built-in web server of PHP][PHPServer]. Please note that PHPs built-in web server is for development and testing purposes only!
+Starting with PHP 5.4 the easiest way to *try* Pico is using PHP's own [built-in web server][PHPServer]. **Please Note** that PHP's built-in web server is for development and testing purposes only!  It is **not** suitable for production use.
+
+* If you decide you like Pico, but need help setting up a more permanent solution, please see our [some section below about server configuration]().
+* Also, only mentioning PHP's internal server seems a bit like a wasted page.  Do we even need a "Run" section?
```
Comment (PhrozenByte) [inline _docs/run.md]: It should be sufficient to link to more in-depth pages for Apache, nginx, PHP's built-in webserver and maybe others here...

File: _docs/customization.md (hunk 1 above)
Comment (smcdougall) [inline _docs/customization.md]: > You really should use different templates for different content categories

*cough* :unamused:

I know though.  Eventually I'd like to break NotePaper up into smaller, more maintainable chunks.  It's already hit that point of being difficult to understand.

I probably should have built it differently from the start.

I do think that this section (or its rewritten equivalent) should go into more detail of how to leverage different templates though.

**Comment by smcdougall** [comment]
> You can btw cherry pick the additional commits on a clean branch

Interesting.  I'll check it out at some point.  Git is always surprising me.  I'm a Git wizard compared to six months ago, and I've still only scratched the surface of what it can do.

File: _docs/customization.md (hunk 1 above)
Comment (PhrozenByte) [inline _docs/customization.md]: Actually I didn't want to say that you should change NotePaper, I meant it completely independent of NotePaper. :wink: For an existing theme this approach is completely fine.

However, we shouldn't recommend this approach for new themes.

Comment (smcdougall) [inline _docs/customization.md]: I know you didn't mean it like that.

It's true though. :laughing:

**Comment by PhrozenByte** [comment]
In response to https://github.com/picocms/Pico/pull/352#issuecomment-225431294:

> I was thinking of not having a main page anymore. [...] Each document would have it's own ToC subsections breaking things down, but not replace the global ToC.

Ah, I see. Sounds like how readthedocs is organized. I'm completely fine with that. I look forward to your suggestions :smiley:

**Comment by smcdougall** [comment]
https://github.com/picocms/Pico/pull/334#issuecomment-255223633 that's a good cookbook example :wink:

**Comment by PhrozenByte** [comment]
Done, see https://github.com/picocms/Pico/compare/d222e809079a1c4b05225d093615d30898b1afa3...cec55eaccad757733f65718a6849a49522d5315b :smiley:

**Comment by smcdougall** [comment]
Lol, okay then. :laughing: :+1:

**Comment by smcdougall** [comment]
Had a rough couple of weeks, but I've been able to get a little work done.

I'll open a new PR sometime soon so we can discuss this over on [picocms.github.io](https://github.com/picocms/picocms.github.io).

(Plus, I've scrapped this branch.  It was only meant to be notes.  I've started over, and I'm just using the old branch as a reference.)

First page of the rewrite is "content complete", though I'm sure there will be lots of revisions and edits by the time everything else is done.

https://smcdougall.github.io/picocms.github.io/installing/

You can leave feedback if you want, but I'm mostly just sharing.  😉

If you want to provide more *general* feedback, such as on the direction I'm going, you can.  The more specific stuff can probably wait though, as it's obviously still very much a work-in-progress.

I won't be around again until tomorrow though.

If you get the chance though, give me a 👍 or 👎. 😄

**Comment by PhrozenByte** [comment]
Nice to hear that! :smiley:

Afaik we've already discussed the general direction (however, I'm not sure *where* :smile:). Just a organizational note: Please move all individual pages to a separate directory (e.g. `docs`; finally a new feature without Jekyll Collections :laughing:). The first page looks good! :+1:

> If you get the chance though, give me a :+1: or :-1:. :smile:

:+1:

:laughing:

**Comment by smcdougall** [comment]
> Please move all individual pages to a separate directory

They will be.

I actually never intended to commit them there.  I recently (finally) changed my server over to Arch, and at the moment I don't have Jekyll installed.  So instead, I started committing the changes to view on GitHub Pages. 😆

I'll reorganize them as I get farther along. 😅

**Comment by PhrozenByte** [comment]
Moved to picocms/picocms.github.io#6...

**Pull request closed without merging**